aerogear/graphback
# Db migrations

## Motivation

This change enables iterative database schema upgrades.

## Tasks

- [x] Add config option to support schema migrations.
- [x] Add new tables to schema.
- [x] Add new fields to existing table in schema.
- [x] Drop database if migrations is turned off.
- [ ] Add migration execution to runtime.

## CLI Verification Steps

1. Add your schema to the models folder (or a custom location if you have changed it in `graphback.json`.
2. In `graphback.json`, set `db.initialization` to "UpdateDatabase".
3. Run `graphback db`.
4. Run `npm run develop`.
5. Add some data to the database.
6. Add a new field to one of your tables.
7. Add a new table.
8. Run `graphback db`.
9. Run `npm run develop`.
10. Check your database. It should have the new fields and new table(s), along with the old data you added in step 4.
11. In `graphback.json`, set `db.initialization` to "DropCreateDatabase".
12. Run `graphback db`. The schema should be recreated from scratch.

## Runtime Verification

`// TODO`
